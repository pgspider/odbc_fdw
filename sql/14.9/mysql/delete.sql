--
-- mysql
-- DELETE
--
--
\set ECHO none
\ir sql/configs/MySql_parameters.conf
\set ECHO all
\i sql/14.9/delete.sql
