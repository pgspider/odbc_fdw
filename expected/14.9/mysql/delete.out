--
-- mysql
-- DELETE
--
--
\set ECHO none
\i sql/14.9/delete.sql
--Testcase 1:
CREATE EXTENSION :DB_EXTENSIONNAME;
--Testcase 2:
CREATE SERVER :DB_SERVERNAME FOREIGN DATA WRAPPER :DB_EXTENSIONNAME
  OPTIONS (odbc_DRIVER :DB_DRIVERNAME, odbc_SERVER :DB_SERVER, odbc_port :DB_PORT, odbc_DATABASE :DB_DATABASE);
--Testcase 3:
CREATE USER MAPPING FOR public SERVER :DB_SERVERNAME OPTIONS (odbc_UID :DB_USER, odbc_PWD :DB_PASS);  
--Testcase 4:
CREATE FOREIGN TABLE delete_test (
    id serial OPTIONS (key 'true'),
    a INT,
    b text
) SERVER :DB_SERVERNAME OPTIONS (schema :DB_SCHEMA, table 'delete_test');
INSERT INTO delete_test (a) VALUES (10);
INSERT INTO delete_test (a, b) VALUES (50, repeat('x', 10000));
INSERT INTO delete_test (a) VALUES (100);
-- allow an alias to be specified for DELETE's target table
--Testcase 5:
EXPLAIN VERBOSE DELETE FROM delete_test AS dt WHERE dt.a > 75;
                                    QUERY PLAN                                    
----------------------------------------------------------------------------------
 Delete on public.delete_test dt  (cost=25.00..28.00 rows=0 width=0)
   Remote SQL: DELETE FROM odbc_fdw_regress.delete_test WHERE id = ?
   ->  Foreign Scan on public.delete_test dt  (cost=25.00..28.00 rows=3 width=4)
         Output: id
         Foreign Table Size: 3 b
         Remote SQL: SELECT id FROM odbc_fdw_regress.delete_test WHERE ((a > 75))
(6 rows)

--Testcase 6:
DELETE FROM delete_test AS dt WHERE dt.a > 75;
-- if an alias is specified, don't allow the original table name
-- to be referenced
--Testcase 7:
EXPLAIN VERBOSE DELETE FROM delete_test dt WHERE delete_test.a > 25;
psql:sql/14.9/delete.sql:29: ERROR:  invalid reference to FROM-clause entry for table "delete_test"
LINE 1: EXPLAIN VERBOSE DELETE FROM delete_test dt WHERE delete_test...
                                                         ^
HINT:  Perhaps you meant to reference the table alias "dt".
--Testcase 8:
DELETE FROM delete_test dt WHERE delete_test.a > 25;
psql:sql/14.9/delete.sql:31: ERROR:  invalid reference to FROM-clause entry for table "delete_test"
LINE 1: DELETE FROM delete_test dt WHERE delete_test.a > 25;
                                         ^
HINT:  Perhaps you meant to reference the table alias "dt".
--Testcase 9:
EXPLAIN VERBOSE SELECT id, a, char_length(b) FROM delete_test;
                               QUERY PLAN                                
-------------------------------------------------------------------------
 Foreign Scan on public.delete_test  (cost=25.00..27.00 rows=2 width=12)
   Output: id, a, char_length(b)
   Foreign Table Size: 2 b
   Remote SQL: SELECT id, a, b FROM odbc_fdw_regress.delete_test
(4 rows)

--Testcase 10:
SELECT id, a, char_length(b) FROM delete_test;
 id | a  | char_length 
----+----+-------------
  1 | 10 |            
  2 | 50 |       10000
(2 rows)

-- delete a row with a TOASTed value
--Testcase 11:
EXPLAIN VERBOSE DELETE FROM delete_test WHERE a > 25;
                                    QUERY PLAN                                    
----------------------------------------------------------------------------------
 Delete on public.delete_test  (cost=25.00..27.00 rows=0 width=0)
   Remote SQL: DELETE FROM odbc_fdw_regress.delete_test WHERE id = ?
   ->  Foreign Scan on public.delete_test  (cost=25.00..27.00 rows=2 width=4)
         Output: id
         Foreign Table Size: 2 b
         Remote SQL: SELECT id FROM odbc_fdw_regress.delete_test WHERE ((a > 25))
(6 rows)

--Testcase 12:
DELETE FROM delete_test WHERE a > 25;
--Testcase 13:
EXPLAIN VERBOSE SELECT id, a, char_length(b) FROM delete_test;
                               QUERY PLAN                                
-------------------------------------------------------------------------
 Foreign Scan on public.delete_test  (cost=25.00..26.00 rows=1 width=12)
   Output: id, a, char_length(b)
   Foreign Table Size: 1 b
   Remote SQL: SELECT id, a, b FROM odbc_fdw_regress.delete_test
(4 rows)

--Testcase 14:
SELECT id, a, char_length(b) FROM delete_test;
 id | a  | char_length 
----+----+-------------
  1 | 10 |            
(1 row)

--Testcase 15:
DROP FOREIGN TABLE delete_test;
--Testcase 16:
DROP SERVER :DB_SERVERNAME CASCADE;
psql:sql/14.9/delete.sql:50: NOTICE:  drop cascades to user mapping for public on server mysql_server
--Testcase 17:
DROP EXTENSION :DB_EXTENSIONNAME CASCADE;
